pageserver/src/tenant/storage_layer/image_layer.rs

```diff
@@ -526,8 +533,17 @@ impl ImageLayerWriterInner {
     ///
     async fn put_image(&mut self, key: Key, img: &[u8]) -> anyhow::Result<()> {
         ensure!(self.key_range.contains(&key));
-        let off = self.blob_writer.write_blob(img).await?;
-
+        let off;
+        if is_rel_data_key(key) {
+            let compressed = lz4_flex::block::compress(img);
+            if compressed.len() < img.len() {
+                off = self.blob_writer.write_blob(&compressed).await?;
+            } else {
+                off = self.blob_writer.write_blob(img).await?;
+            }
+        } else {
+            off = self.blob_writer.write_blob(img).await?;
```

Comment:
What are these images, are they the internal values like key 0 file listing description we output?

Reply:
Compression is only performed for relation pages (8kb blocks)